### PR TITLE
Secrets management documentation for Buildbot

### PR DESCRIPTION
**Pull request by protatremy**
How to use the secrets providers defined with secrets defined in
a file system and secrets wrote in Vault backend

### CODE REVIEW

File: master/docs/manual/secretsmanagement.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
+Secret Management
+=================
+
+Buildbot uses secrets when it is executing tasks during a build.
```
Comment (tardyp) [inline master/docs/manual/secretsmanagement.rst]: >> Buildbot steps might need secrets to execute their actions.

File: master/docs/manual/secretsmanagement.rst [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+Secrets are used to execute commands or remote connections.
```
Comment (tardyp) [inline master/docs/manual/secretsmanagement.rst]: > Secrets can be used to execute commands or to create authenticated network connections.

File: master/docs/manual/secretsmanagement.rst (hunk 1 above)
Comment (protatremy) [inline master/docs/manual/secretsmanagement.rst]: replaced

File: master/docs/manual/secretsmanagement.rst (hunk 2 above)
Comment (protatremy) [inline master/docs/manual/secretsmanagement.rst]: corrected

**Comment by tardyp** [comment]
looks like you messed up your rebase :o)

**Comment by protatremy** [comment]
yes my rebase was totally messed up, sorry for the trouble...

File: master/docs/developer/secrets.rst [hunk 3]
```diff
@@ -1,6 +1,10 @@
 Secrets
 -------
 
+.. spelling::
+
+  secretDetails
```
Comment (rutsky) [inline master/docs/developer/secrets.rst]: Please wrap `secretDetails` in double backticks instead --- it's code, not English sentence:

```
return a list of ``secretDetails``.
```

**Comment by Frodox** [comment]
I really like the idea, but I'm a bit confused by examples..:)

File: master/docs/manual/secretsmanagement.rst [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+- File system based: secrets are written in a file.
+  This is a simple solution for example when secrets are managed by config management system like Ansible Vault.
+
+- Third party backend based: secrets are stored by a specialized software.
+  These solution are usually more secured.
+
+Secrets providers are configured if needed in the master configuration.
+Multiple providers can be configured at once.
+The secret manager is a Buildbot service. The secret manager returns the specific provider results related to the providers registered in the configuration.
```
Comment (seankelly) [inline master/docs/manual/secretsmanagement.rst]: Put the second sentence on a new line.

File: master/docs/manual/secretsmanagement.rst [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+How to use secrets in Buildbot
+------------------------------
+
+The following example shows a basic usage of secrets in buildbot
```
Comment (seankelly) [inline master/docs/manual/secretsmanagement.rst]: Buildbot* and add a period at the end.

File: master/docs/manual/secretsmanagement.rst [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+.. code-block:: python
+
+    # First we declare that the secrets are stored in a directory of the filesystem
+    # each file contain one secret identified by the filename
+    c['secretsProviders'] = [util.SecretInFile(directory="/path/toSecretsFiles"]
+
+    # then in a buildfactory:
+
+    #  use a secret on a shell command via Interpolate
```
Comment (seankelly) [inline master/docs/manual/secretsmanagement.rst]: Remove a space after the `#`.

File: master/docs/manual/secretsmanagement.rst [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+`````````````
+
+.. code-block:: python
+
+    c['secretsProviders'] = [util.SecretInVault(
+                            vaultToken=open('VAULT_TOKEN').read(),
+                            vaultServer="http://localhost:8200"
+    )]
+
+Vault secures, stores, and tightly controls access to secrets. Vault presents an unified API to access multiple backends.
```
Comment (seankelly) [inline master/docs/manual/secretsmanagement.rst]: Put second sentence on a new line.

File: master/docs/manual/secretsmanagement.rst [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+To be authenticated in Vault, Buildbot need to send a token to the vault server.
+The token is generated when the Vault instance is initialized for the first time.
+
+
+In the master configuration the Vault provider is instantiated trough the Buildbot service manager as a secret provider with the the Vault server address and the Vault token.
```
Comment (seankelly) [inline master/docs/manual/secretsmanagement.rst]: trough -> through

Also add a comma after "configuration".

File: master/docs/manual/secretsmanagement.rst [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+How to configure a Vault instance
+---------------------------------
+
+Vault being a very generic system, it can be complex to install for the first time.
+Here is a simple tutorial to install the minimal Vault for use with Buildbot.
+
+Use Docker to install Vault
+```````````````````````````
+
+A Docker is available to help users installing Vault.
```
Comment (seankelly) [inline master/docs/manual/secretsmanagement.rst]: A Docker image*

File: master/docs/manual/secretsmanagement.rst [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+Without argument the command launch a docker vault developer instance, easy to use and test the functions.
```
Comment (seankelly) [inline master/docs/manual/secretsmanagement.rst]: "Without any arguments, the command launches a Docker Vault..."

File: master/docs/manual/secretsmanagement.rst [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+In a shell:
+
+.. code-block:: shell
+
+    docker run vault
+
+Starting the vault instance
+```````````````````````````
+
+Once the docker image is created, launch a shell terminal on the docker image:
```
Comment (seankelly) [inline master/docs/manual/secretsmanagement.rst]: Docker* both instances.

File: master/docs/manual/secretsmanagement.rst [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+
+      docker exec -i -t ``docker_vault_image_name`` /bin/sh
+
+Then, export the environment variable VAULT_ADDR needed to init Vault.
+
+.. code-block:: shell
+
+      export VAULT_ADDR='vault.server.adress'
+
+Writing a secrets
```
Comment (seankelly) [inline master/docs/manual/secretsmanagement.rst]: Writing secrets*

Comment (protatremy) [inline master/docs/manual/secretsmanagement.rst]: Thanks a lot for your review seankelly!

**Comment by tardyp** [comment]
@seankelly I'll let you do the final merge

**Pull request merged**